hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Event.java

```diff
@@ -57,51 +67,105 @@
     // This field will be controlled vi fasterxml. We need to make this an explicit field because we ship the
     // serialized json to clients via rest. Since we serialize via fasterxml the field will be added to the json.
     // Using "visible=false" in JsonTypeInfo only works if we control the deserialization.
+    @ApiModelProperty(value = "Indicate if this object is an EVENT or an ALERT. This is a read-only field controlled " +
+            "by the system for serialization purposes.",
+            position = 0,
+            allowableValues = "EVENT, ALERT")
     @JsonInclude
     protected String eventType;
 
+    @ApiModelProperty(value = "Tenant id owner of this event.",
+            position = 1,
+            allowableValues = "Tenant is overwritten from Hawkular-Tenant HTTP header parameter request")
     @JsonInclude
     protected String tenantId;
 
+    @ApiModelProperty(value = "Unique identifier for this event.",
+            position = 2,
+            required = true)
     @JsonInclude
     protected String id;
 
+    @ApiModelProperty(value = "Creation time for this event.",
+            position = 3,
+            required = true,
+            allowableValues = "Timestamp in milliseconds.")
     @JsonInclude
     protected long ctime;
 
     // Optional dataSource of Event, used by EventCondition to evaluate events
+    @ApiModelProperty(value = "Optional dataSource for Event. Used for <<Trigger>> in <<EventCondition>> " +
+            "to evaluate events with triplet [tenantId, source, dataId] as unique identifier.",
+            position = 4,
+            allowableValues = "Timestamp in milliseconds.")
     @JsonInclude
     String dataSource;
 
     // Optional dataId of Event, used by EventCondition to evaluate events
+    @ApiModelProperty(value = "Data identifier used for Events condition evaluation. All Events participating on " +
+            "conditions must have a valid dataId. DataId in events context can be read as an events source name. + \n" +
+            "Events generated from <<Trigger>> will use the triggerId as dataId.",
```

Comment:
Suggested:
'Data identifier used for Events condition evaluation. All Events participating on conditions must have a valid dataId. DataId in events context can be read as an events source name. Events generated from <<Trigger>> will use the triggerId as dataId.'
-->
'Data identifier used for Event condition evaluation.  Events must supply a valid dataId to be considered for EventCondition evaluation.  DataIds in an events context should incorporate the source of the event (for uniqueness). Events generated from a <<Trigger>> will have dataId set to the triggerId, therefore allowing chaining with other EventConditions.'

If you don't take the suggestion you must minimally: 'events' -> 'event's where used in the possessive sense.